util/types/time.go

```diff
@@ -437,6 +437,15 @@ func (t *Time) check() error {
 	return nil
 }
 
+// TotalDays returns number of days starting from 0/1/1
+func (t Time) TotalDays() int {
```

Comment:
In mysql, 2016-00-13 is a valid date, not an error. You can store it, read it back, pass it as argument in many functions...
We are struggle with that "feature", because the promise of mysql compatibility.